bob-apps/bin/bobp.rs

```diff
@@ -470,113 +592,117 @@ async fn stat_worker(
     keys_count: u64,
     behavior_flags: u8,
 ) {
-    let (put_speed_values, get_speed_values, exist_speed_values, elapsed) =
+    let (put_speed_values, get_speed_values, exist_speed_values, delete_speed_values, elapsed) =
         print_periodic_stat(stop_token, period_ms, &stat, request_bytes, behavior_flags);
-    print_averages(
-        &stat,
-        &put_speed_values,
-        &get_speed_values,
-        &exist_speed_values,
-        elapsed,
-        keys_count,
-        behavior_flags,
-    );
-    print_errors_with_codes(stat);
+    print_averages(&stat, &put_speed_values, &get_speed_values, &exist_speed_values, &delete_speed_values, elapsed, keys_count, behavior_flags);
+    print_errors_with_codes(stat).await
 }
 
-fn print_errors_with_codes(stat: Arc<Statistics>) {
-    let guard = stat.get_errors.lock().expect("mutex");
+async fn print_errors_with_codes(stat: Arc<Statistics>) {
+    let guard = stat.get.common.errors.lock().expect("mutex");
     if !guard.is_empty() {
         println!("get errors:");
         for (&code, count) in guard.iter() {
             println!("{:?} = {}", Code::from(code), count);
         }
     }
-    let guard = stat.put_errors.lock().expect("mutex");
+    let guard = stat.put.common.errors.lock().expect("mutex");
     if !guard.is_empty() {
         println!("put errors:");
         for (&code, count) in guard.iter() {
             println!("{:?} = {}", Code::from(code), count);
         }
     }
-    let guard = stat.exist_errors.lock().expect("mutex");
+    let guard = stat.exist.common.errors.lock().expect("mutex");
     if !guard.is_empty() {
         println!("exist errors:");
         for (&code, count) in guard.iter() {
             println!("{:?} = {}", Code::from(code), count);
         }
     }
+    let guard = stat.delete.common.errors.lock().expect("mutex");
```

Comment:
4 identical blocks of code. Declare function `print_errors` in `OperationStatistics` that receives operation name as parameter